### ISSUE TITLE
[go_router_builder] go() get null on extra previous route

### ISSUE THREAD

**Issue by CaoGiaHieu-dev** [flutter/flutter#147413]
### Steps to reproduce

Create router follow this 
```
@TypedStatefulShellRoute<DashboardRoute>(
  branches: [
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<HomeRoute>(
          path: Routes.home,
          routes: [
            TypedGoRoute<PostDetailRoute>(
              path: Routes.postDetail,
            )
          ],
        ),
      ],
    ),
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<ApplyRoute>(
          path: Routes.apply,
        ),
      ],
    ),
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<ChatRoute>(
          path: Routes.chat,
        ),
      ],
    ),
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<ProfileRoute>(
          path: Routes.profile,
          routes: [
            TypedGoRoute<SettingRoute>(
              path: Routes.setting,
              routes: [
                TypedGoRoute<NoticeRoute>(
                  path: Routes.notice,
                ),
                TypedGoRoute<AlarmRoute>(
                  path: Routes.alarmSetting,
                ),
                TypedGoRoute<TermOfConditionRoute>(
                  path: Routes.termOfCondition,
                ),
                TypedGoRoute<PrivacyRoute>(
                  path: Routes.privacyPolicy,
                ),
                TypedGoRoute<LanguageRoute>(
                  path: Routes.language,
                ),
              ],
            ),
            TypedGoRoute<AppliedPostDetailRoute>(
              path: Routes.appliedPostDetail,
            ),
          ],
        ),
      ],
    ),
  ],
)
```
Navigate to Routes.postDetail with JobsExtra $extra and using go to  Routes.chat by const ChatRoute().go(context);

### Expected results

Navigate success

### Actual results

got error type 'Null' is not a subtype of type 'JobsExtra' in type cast

### Code sample

<details open><summary>Code sample</summary>

```dart
// create router
@TypedStatefulShellRoute<DashboardRoute>(
  branches: [
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<HomeRoute>(
          path: Routes.home,
          routes: [
            TypedGoRoute<PostDetailRoute>(
              path: Routes.postDetail,
            )
          ],
        ),
      ],
    ),
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<ApplyRoute>(
          path: Routes.apply,
        ),
      ],
    ),
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<ChatRoute>(
          path: Routes.chat,
        ),
      ],
    ),
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<ProfileRoute>(
          path: Routes.profile,
          routes: [
            TypedGoRoute<SettingRoute>(
              path: Routes.setting,
              routes: [
                TypedGoRoute<NoticeRoute>(
                  path: Routes.notice,
                ),
                TypedGoRoute<AlarmRoute>(
                  path: Routes.alarmSetting,
                ),
                TypedGoRoute<TermOfConditionRoute>(
                  path: Routes.termOfCondition,
                ),
                TypedGoRoute<PrivacyRoute>(
                  path: Routes.privacyPolicy,
                ),
                TypedGoRoute<LanguageRoute>(
                  path: Routes.language,
                ),
              ],
            ),
            TypedGoRoute<AppliedPostDetailRoute>(
              path: Routes.appliedPostDetail,
            ),
          ],
        ),
      ],
    ),
  ],
)
```
Navigate to PostDetailRoute with $extra then call const ChatRoute().go(context); 

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
#4      GoRouterDelegate._callOnExitStartsAt.handleOnExitResult (package:go_router/src/delegate.dart:275:16)
delegate.dart:275
#5      GoRouterDelegate._callOnExitStartsAt (package:go_router/src/delegate.dart:290:14)
delegate.dart:290
#6      GoRouterDelegate.setNewRoutePath (package:go_router/src/delegate.dart:232:16)
delegate.dart:232
#7      _RouterState._processParsedRouteInformation.<anonymous closure> (package:flutter/src/widgets/router.dart:758:34)
router.dart:758
#8      SynchronousFuture.then (package:flutter/src/foundation/synchronous_future.dart:43:39)
synchronous_future.dart:43
#9      _RouterState._processRouteInformation (package:flutter/src/widgets/router.dart:750:8)
router.dart:750
#10     _RouterState._handleRouteInformationProviderNotification (package:flutter/src/widgets/router.dart:767:5)
router.dart:767
#11     ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:433:24)
change_notifier.dart:433
#12     GoRouteInformationProvider.notifyListeners (package:go_router/src/information_provider.dart:134:11)
information_provider.dart:134
#13     GoRouteInformationProvider._setValue (package:go_router/src/information_provider.dart:144:7)
information_provider.dart:144
#14     GoRouteInformationProvider.go (package:go_router/src/information_provider.dart:166:5)
information_provider.dart:166
#15     GoRouter.go (package:go_router/src/router.dart:340:30)
router.dart:340
#16     GoRouterHelper.go (package:go_router/src/misc/extensions.dart:25:25)
extensions.dart:25
#17     $ChatRouteExtension.go (package:sayyo/router/app_routes.g.dart:195:44)
app_routes.g.dart:195
#18     ChatWidget.build.<anonymous closure> (package:sayyo/screens/applicants/user_applicant_screen.dart:59:27)
user_applicant_screen.dart:59
#19     _InkResponseState.handleTap (package:flutter/src/material/ink_well.dart:1183:21)
ink_well.dart:1183
#20     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:315:24)
recognizer.dart:315
#21     TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:652:11)
tap.dart:652
#22     BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:309:5)
tap.dart:309
#23     BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:279:7)
tap.dart:279
#24     GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:167:27)
arena.dart:167
#25     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:499:20)
binding.dart:499
#26     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:475:22)
binding.dart:475
#27     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:430:11)
binding.dart:430
#28     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:420:7)
binding.dart:420
#29     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:383:5)
binding.dart:383
#30     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:330:7)
binding.dart:330
#31     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:299:9)
binding.dart:299
#35     _invoke1 (dart:ui/hooks.dart:330:10)
hooks.dart:330
#36     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:429:7)
platform_dispatcher.dart:429
#37     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)
hooks.dart:262
(elided 3 frames from dart:async)
════════════════════════════════════════════════════════════════════════════════
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
flutter doctor -v
[✓] Flutter (Channel stable, 3.19.6, on macOS 14.4.1 23E224 darwin-x64, locale vi)
    • Flutter version 3.19.6 on channel stable at /Users/caogiahieu/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 54e66469a9 (9 days ago), 2024-04-17 13:08:03 -0700
    • Engine revision c4cd48e186
    • Dart version 3.3.4
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/caogiahieu/Library/Android/sdk
    • Platform android-34, build-tools 31.0.0
    • ANDROID_HOME = /Users/caogiahieu/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/caogiahieu/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Android Studio (version 2023.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.88.1)
    • VS Code at /Users/caogiahieu/Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.86.0

[✓] Connected device (1 available)
    • iPhone của Cao Gia Hieu (mobile) • e3010bcca1500cf8a43f8baba42c0d712114c722 • ios • iOS 16.7.7 20H330

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by CaoGiaHieu-dev** [comment]
it work fine if i use 
```
context.pop();
        WidgetsBinding.instance.endOfFrame.whenComplete(() {
          context.pop();
          WidgetsBinding.instance.endOfFrame.whenComplete(() {
            const ChatRoute().go(context);
          });
        });
```

**Comment by darshankawar** [comment]
Thanks for the report. I was able to replicate the reported behavior / exception.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.19.6, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.19.6 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 54e66469a9 (6 days ago), 2024-04-17 13:08:03 -0700
    • Engine revision c4cd48e186
    • Dart version 3.3.4
    • DevTools version 2.31.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.22.0-18.0.pre.53, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.22.0-18.0.pre.53 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 098e7e7ff3 (4 hours ago), 2024-04-29 01:25:19 +0000
    • Engine revision 752b146df7
    • Dart version 3.5.0 (build 3.5.0-109.0.dev)
    • DevTools version 2.35.0-dev.16
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by CaoGiaHieu-dev** [comment]
Any update one this issue ?

**Comment by chunhtai** [comment]
@darshankawar what is the full repro are you using? the example code in this issue is not complete

**Comment by darshankawar** [comment]
Sorry for inconvenience @chunhtai. I somehow missed to post it here while verifying. I'll share it here shortly.

**Comment by chunhtai** [comment]
Hi @darshankawar have you had the time to look into the setup?

**Comment by darshankawar** [comment]
Sorry I was out of office so couldn't post it earlier. 
When I verified this issue, I used the code snippet provided by OP and used it in plugin example's  [file](https://github.com/flutter/packages/blob/main/packages/go_router_builder/example/lib/stateful_shell_route_example.dart).
But after that, trying to run the same code doesn't seem to trigger the reported behavior at my end.
 
Below is the complete code I had used:

<details>
<summary> runnable code </summary>

```

// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// ignore_for_file: public_member_api_docs, unreachable_from_main

import 'package:collection/collection.dart';
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';

part 'stateful_shell_route_example.g.dart';

final GlobalKey<NavigatorState> _sectionANavigatorKey =
    GlobalKey<NavigatorState>(debugLabel: 'sectionANav');

void main() => runApp(App());

class App extends StatelessWidget {
  App({super.key});

  @override
  Widget build(BuildContext context) => MaterialApp.router(
        routerConfig: _router,
      );

  final GoRouter _router = GoRouter(
    routes: $appRoutes,
    initialLocation: '/detailsA',
  );
}

class HomeScreen extends StatelessWidget {
  const HomeScreen({super.key});

  @override
  Widget build(BuildContext context) => Scaffold(
        appBar: AppBar(title: const Text('foo')),
      );
}

// create router
@TypedStatefulShellRoute<MyShellRouteData>(
  branches: [
    TypedStatefulShellBranch(
      routes: [
        TypedGoRoute<DetailsARouteData>(
          path: '/detailsA',
          routes: [
            TypedGoRoute<DetailsBRouteData>(
              path: '/detailsB',
            )
          ],
        ),
      ],
    ),
          ],
)
class MyShellRouteData extends StatefulShellRouteData {
  const MyShellRouteData();

  @override
  Widget builder(
    BuildContext context,
    GoRouterState state,
    StatefulNavigationShell navigationShell,
  ) {
    return navigationShell;
  }

  static const String $restorationScopeId = 'restorationScopeId';

  static Widget $navigatorContainerBuilder(BuildContext context,
      StatefulNavigationShell navigationShell, List<Widget> children) {
    return ScaffoldWithNavBar(
      navigationShell: navigationShell,
      children: children,
    );
  }
}

class BranchAData extends StatefulShellBranchData {
  const BranchAData();
}

class BranchBData extends StatefulShellBranchData {
  const BranchBData();

  static final GlobalKey<NavigatorState> $navigatorKey = _sectionANavigatorKey;
  static const String $restorationScopeId = 'restorationScopeId';
}

class DetailsARouteData extends GoRouteData {
  const DetailsARouteData();

  @override
  Widget build(BuildContext context, GoRouterState state) {
    return const DetailsScreen(label: 'A');
  }
}

class DetailsBRouteData extends GoRouteData {
  const DetailsBRouteData();

  @override
  Widget build(BuildContext context, GoRouterState state) {
    return const DetailsScreen(label: 'B');
  }
}

/// Builds the "shell" for the app by building a Scaffold with a
/// BottomNavigationBar, where [child] is placed in the body of the Scaffold.
class ScaffoldWithNavBar extends StatelessWidget {
  /// Constructs an [ScaffoldWithNavBar].
  const ScaffoldWithNavBar({
    required this.navigationShell,
    required this.children,
    Key? key,
  }) : super(key: key ?? const ValueKey<String>('ScaffoldWithNavBar'));

  /// The navigation shell and container for the branch Navigators.
  final StatefulNavigationShell navigationShell;

  /// The children (branch Navigators) to display in a custom container
  /// ([AnimatedBranchContainer]).
  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: AnimatedBranchContainer(
        currentIndex: navigationShell.currentIndex,
        children: children,
      ),
      bottomNavigationBar: BottomNavigationBar(
        // Here, the items of BottomNavigationBar are hard coded. In a real
        // world scenario, the items would most likely be generated from the
        // branches of the shell route, which can be fetched using
        // `navigationShell.route.branches`.
        items: const <BottomNavigationBarItem>[
          BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Section A'),
          BottomNavigationBarItem(icon: Icon(Icons.work), label: 'Section B'),
        ],
        currentIndex: navigationShell.currentIndex,
        onTap: (int index) => _onTap(context, index),
      ),
    );
  }

  /// Navigate to the current location of the branch at the provided index when
  /// tapping an item in the BottomNavigationBar.
  void _onTap(BuildContext context, int index) {
    // When navigating to a new branch, it's recommended to use the goBranch
    // method, as doing so makes sure the last navigation state of the
    // Navigator for the branch is restored.
    navigationShell.goBranch(
      index,
      // A common pattern when using bottom navigation bars is to support
      // navigating to the initial location when tapping the item that is
      // already active. This example demonstrates how to support this behavior,
      // using the initialLocation parameter of goBranch.
      initialLocation: index == navigationShell.currentIndex,
    );
  }
}

/// Custom branch Navigator container that provides animated transitions
/// when switching branches.
class AnimatedBranchContainer extends StatelessWidget {
  /// Creates a AnimatedBranchContainer
  const AnimatedBranchContainer(
      {super.key, required this.currentIndex, required this.children});

  /// The index (in [children]) of the branch Navigator to display.
  final int currentIndex;

  /// The children (branch Navigators) to display in this container.
  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    return Stack(
        children: children.mapIndexed(
      (int index, Widget navigator) {
        return AnimatedScale(
          scale: index == currentIndex ? 1 : 1.5,
          duration: const Duration(milliseconds: 400),
          child: AnimatedOpacity(
            opacity: index == currentIndex ? 1 : 0,
            duration: const Duration(milliseconds: 400),
            child: _branchNavigatorWrapper(index, navigator),
          ),
        );
      },
    ).toList());
  }

  Widget _branchNavigatorWrapper(int index, Widget navigator) => IgnorePointer(
        ignoring: index != currentIndex,
        child: TickerMode(
          enabled: index == currentIndex,
          child: navigator,
        ),
      );
}

/// The details screen for either the A or B screen.
class DetailsScreen extends StatefulWidget {
  /// Constructs a [DetailsScreen].
  const DetailsScreen({
    required this.label,
    this.param,
    this.extra,
    super.key,
  });

  /// The label to display in the center of the screen.
  final String label;

  /// Optional param
  final String? param;

  /// Optional extra object
  final Object? extra;
  @override
  State<StatefulWidget> createState() => DetailsScreenState();
}

/// The state for DetailsScreen
class DetailsScreenState extends State<DetailsScreen> {
  int _counter = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Details Screen - ${widget.label}'),
      ),
      body: _build(context),
    );
  }

  Widget _build(BuildContext context) {
    return Center(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: <Widget>[
          Text('Details for ${widget.label} - Counter: $_counter',
              style: Theme.of(context).textTheme.titleLarge),
          const Padding(padding: EdgeInsets.all(4)),
          TextButton(
            onPressed: () {
              setState(() {
                _counter++;
              });
            },
            child: const Text('Increment counter'),
          ),
          const Padding(padding: EdgeInsets.all(8)),
          if (widget.param != null)
            Text('Parameter: ${widget.param!}',
                style: Theme.of(context).textTheme.titleMedium),
          const Padding(padding: EdgeInsets.all(8)),
          if (widget.extra != null)
            Text('Extra: ${widget.extra!}',
                style: Theme.of(context).textTheme.titleMedium),
        ],
      ),
    );
  }
}

```

</details>

@CaoGiaHieu-dev If you could provide us latest updated code that still triggers the error, would be helpful to Chunhtai to investigate further.

**Comment by CaoGiaHieu-dev** [comment]
Seems like it was fixed on  go_router_builder : ^2.7.1 and go_router: ^14.2.7 .
